src/markdown-to-inner-partitions.js

```diff
@@ -0,0 +1,214 @@
+      partitions.push(_buildTextPartition(markdownSubstring.substring(currentIndex, nextIndex)));
+      currentIndex = nextIndex;
+    } else {
+      currentChar = markdownSubstring.charAt(currentIndex);
+      if (currentChar === '*') {
+        close = _findNextValidChar(currentChar, currentIndex + 1, markdownSubstring);
+        if (close > 0) {
+          partitions.push(_buildBoldPartition(markdownSubstring.substring(currentIndex + 1, close)));
+        }
+      } else if (currentChar === '_') {
+        close = _findNextValidChar(currentChar, currentIndex + 1, markdownSubstring);
+        if (close > 0) {
+          partitions.push(_buildItalicsPartition(markdownSubstring.substring(currentIndex + 1, close)));
+        }
+      } else if (currentChar === '[') {
+        close = markdownSubstring.indexOf(')', currentIndex);
+        partitions.push(_buildLinkPartition(markdownSubstring.substring(currentIndex, close + 1)));
+      } else if (currentChar === '{') {
+        close = markdownSubstring.indexOf(')', currentIndex);
+        partitions.push(_buildRelationOrColorPartition(markdownSubstring.substring(currentIndex, close + 1)));
+      }
+      currentIndex = close + 1;
+    }
+    nextIndex = _findNextInnerStart(currentIndex, markdownSubstring);
+  }
+  if (currentIndex !== nextIndex) {
+    partitions.push(_buildTextPartition(markdownSubstring.substring(currentIndex)));
+  }
+  return partitions;
+}
+
+function _buildBoldPartition(markdownSubstring) {
+  let partitions = generateInnerPartitions(markdownSubstring);
+  if (!partitions) {
+    return { type: PARTITION_TYPES.BOLD, value: markdownSubstring.replace(/\\/g, '') };
+  } else {
+    return { type: PARTITION_TYPES.BOLD, partitions };
+  }
+}
+
+function _buildItalicsPartition(markdownSubstring) {
+  let partitions = generateInnerPartitions(markdownSubstring);
+  if (!partitions) {
+    return { type: PARTITION_TYPES.ITALICS, value: markdownSubstring.replace(/\\/g, '') };
+  } else {
+    return { type: PARTITION_TYPES.ITALICS, partitions };
+  }
+}
+
+function _buildLinkPartition(markdownSubstring) {
+  let breaks = _breakLinkRelationColor(']', markdownSubstring.replace(/\\/g, ''));
+  return { type: PARTITION_TYPES.LINK, value: breaks[0], link: breaks[1] };
+}
+
+function _buildRelationOrColorPartition(markdownSubstring) {
+  let breaks = _breakLinkRelationColor('}', markdownSubstring.replace(/\\/g, ''));
+  if (breaks[2] !== true) {
+    return _buildRelationPartition(breaks);
+  } else {
+    return _buildColorPartition(breaks);
+  }
+}
+
+function _buildRelationPartition(breaks) {
+  return { type: PARTITION_TYPES.RELATION, value: breaks[0], relation: breaks[1] };
+}
+
+function _buildColorPartition(breaks) {
+  return { type: PARTITION_TYPES.COLOR, value: breaks[0], color: breaks[1] };
+}
+
+function _buildTextPartition(markdownSubstring) {
+  return { type: PARTITION_TYPES.TEXT, value: markdownSubstring.replace(/\\/g, '') };
+}
+
+// HELPER METHODS
```

Comment:
Aren't all the `_*` (private) functions helper functions?

Reply:
Yes they are but the functions about the helper comment are for a specific partition. The functions below that are things that do the general (non-partition-specific) work. It made sense to me.